### PR TITLE
Use both variant and bandwidth

### PR DESCRIPTION
**Pull request by colekas-ssim**

### CODE REVIEW

File: libavformat/hls.c [hunk 1]
```diff
@@ -226,12 +226,24 @@ typedef struct HLSContext {
     int variant_count;
 } HLSContext;
 
+// If both index and bandwidth are selected, the ensure both match
```
Comment (ahmedabadr) [inline libavformat/hls.c]: the -> then

File: libavformat/hls.c [hunk 2]
```diff
@@ -226,12 +226,24 @@ typedef struct HLSContext {
...
 static int is_variant_selected(HLSContext* c, const char* current_bandwidth) {
     // If both selected bandwidth and selected variant are default, then add this program
     if (!c->selected_bandwidth && c->selected_variant_index == -1) {
         return 1;
     }
     // If a variant is selected and we've counted enough variants
+    // and the bandwidth is selected
+    else if (c->selected_variant_index != -1 && c->selected_bandwidth &&
```
Comment (ahmedabadr) [inline libavformat/hls.c:233]: if none of them are selected, do we return all variants?

Comment (ahmedabadr) [reply libavformat/hls.c:233]: I think we do .. it's the first condition in the function!

**Review by ahmedabadr — APPROVED** [review]

File: libavformat/hls.c (hunk 2 above)
Comment (colekas-imax) [reply libavformat/hls.c:233]: We do. This way we can do filtering in the application if we want in the future.

**Pull request merged**